cuda_core/cuda/core/experimental/_module.py

```diff
@@ -215,6 +216,36 @@ def attributes(self) -> KernelAttributes:
             self._attributes = KernelAttributes._init(self._handle)
         return self._attributes
 
+    @property
+    def num_arguments(self) -> int:
+        """int : The number of arguments of this function"""
+        attr_impl = self.attributes
+        if attr_impl._backend_version != "new":
+            raise NotImplementedError("New backend is required")
+        arg_pos = 0
+        while True:
+            result = attr_impl._loader["paraminfo"](self._handle, arg_pos)
+            if result[0] != driver.CUresult.CUDA_SUCCESS:
+                break
+            arg_pos = arg_pos + 1
+        return arg_pos
+
+    @property
+    def arguments_info(self) -> list[tuple[int, int]]:
```

Comment:
The two implementations are very similar.

Did you consider using a private helper function?

```python
    def _arguments_helper(self, param_info=None):
        ...
        if param_info is not None:
            param_info.append((results[1], result[2]))
        ...
```

Then:

```python
    @property
    def num_arguments(self) -> int:
        return self._arguments_helper()

    @property
    def arguments_info(self) -> list[tuple[int, int]]:
        param_info = []
        self._arguments_helper(param_info)
        return param_info
```

Reply:
Good idea, thanks @rwgk . I have pushed the change.